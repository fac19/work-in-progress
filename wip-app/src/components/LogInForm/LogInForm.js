import React from "react";
import { Container, Button } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { useHistory } from "react-router-dom";
import { logInPost } from "../../utils/post-fetch";
import { HeaderLogoStyle } from "../../components/Logo.style";
import { FormInput, FormLabel } from "./LogInForm.style";
import { Link } from "react-router-dom";

const useStyles = makeStyles({
  form: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    width: "90%",
  },
  formContainer: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
  },
  formElement: {
    margin: "0.5rem",
  },
  errorMessage: {
    color: "red",
  },
});

const LogInForm = (props) => {
  const [form, setForm] = React.useState({ username: "", password: "" });
  const [error, setError] = React.useState("");

  const classes = useStyles();
  const history = useHistory();

  const handleChange = (event) => {
    let { name, value } = event.target;
    setForm({ ...form, [name]: value });
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    logInPost(form)
      .then(() => history.push("/feed"))
      .catch(() =>
        setError("Could not log you in, your details may be incorrect!")
      );
  };

  return (
    <Container className={classes.formContainer} component="main" maxWidth="xs">
      <HeaderLogoStyle alt="work in progress logo" src="logo.svg" />
      <h1>Log in</h1>
      <form className={classes.form} onSubmit={handleSubmit}>
        <FormLabel htmlFor="username">Username *</FormLabel>
        <FormInput
          type="text"
          id="username"
          placeholder="Username"
          name="username"
          required
          autoFocus
          onChange={handleChange}
        ></FormInput>
        <FormLabel htmlFor="password">Password *</FormLabel>
        <FormInput
          type="password"
          id="password"
          name="password"
          placeholder="Password"
          required
          onChange={handleChange}
        ></FormInput>
        <Button
          className={classes.formElement}
          variant="contained"
          color="secondary"
          type="submit"
        >
          Log In
        </Button>
        <p className={classes.errorMessage}>{error}</p>
      </form>
      <p>
        Don't have an account yet? <Link to="/sign-up">Sign up</Link>
      </p>
    </Container>
  );
};

export default LogInForm;
