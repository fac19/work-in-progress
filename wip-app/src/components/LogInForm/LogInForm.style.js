import styled from "styled-components";

const FormInput = styled.input`
  width: 100%;
  height: 2.5rem;
  border: 1px solid black;
  font-size: 1.1rem;
  margin-bottom: 1.7rem;
`;

const FormLabel = styled.label`
  text-align: left;
  width: 100%;
`;

export { FormInput, FormLabel };
