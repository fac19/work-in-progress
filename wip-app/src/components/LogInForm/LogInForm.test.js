import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import LogInForm from './LogInForm'

test("Jest is working", () => {
  expect(true).toBeTruthy();
});