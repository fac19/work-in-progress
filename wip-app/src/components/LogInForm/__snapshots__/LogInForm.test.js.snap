// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<main
  className="MuiContainer-root makeStyles-formContainer-2 MuiContainer-maxWidthXs"
>
  <img
    alt="work in progress logo"
    className="sc-AxjAm ecCdST"
    src="logo.svg"
  />
  <h1>
    Log in
  </h1>
  <form
    className="makeStyles-form-1"
    onSubmit={[Function]}
  >
    <label
      className="sc-AxhCb djpVYo"
      htmlFor="username"
    >
      Username *
    </label>
    <input
      autoFocus={true}
      className="sc-AxiKw dxRHDb"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      required={true}
      type="text"
    />
    <label
      className="sc-AxhCb djpVYo"
      htmlFor="password"
    >
      Password *
    </label>
    <input
      className="sc-AxiKw dxRHDb"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formElement-3 MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Log In
      </span>
    </button>
    <p
      className="makeStyles-errorMessage-4"
    >
      
    </p>
  </form>
  <p>
    Don't have an account yet? 
    <a
      href="/sign-up"
      onClick={[Function]}
    >
      Sign up
    </a>
  </p>
</main>
`;
